### PR TITLE
Initial functionality

### PR DESCRIPTION
**Pull request by dav3r**
#  #

## 🗣 Description ##

This PR covers the initial functionality required to generate an Assessor Portal AMI that can be used in [the COOL](https://github.com/cisagov/cool-system).

## 💭 Motivation and context ##

This PR is an important part of the work required for https://github.com/cisagov/cool-system/issues/202 and it depends on https://github.com/cisagov/ansible-role-assessor-portal/pull/3.

Note that since the Assessor Portal requires several gigabytes of Docker images to function, those Docker images are pulled in as part of the construction of this AMI (see https://github.com/cisagov/assessor-portal-packer/pull/1/commits/c7a3e80b8240d8a467e85dda7cfaf13a11ef0253, https://github.com/cisagov/assessor-portal-packer/pull/1/commits/90b76b5905fd8224027bf46928352320f5e437ee, and https://github.com/cisagov/assessor-portal-packer/pull/1/commits/a2119926505aaa36fd8023a035036e318debc5da).  This is mainly to avoid having to fetch those Docker images every time a fresh Assessor Portal instance is deployed.

## 🧪 Testing ##

This AMI was tested in conjunction with https://github.com/cisagov/cool-assessment-terraform/pull/130, which adds support for these Assessor Portal instances.

Testing confirmed all of the following:
- [x] An Assessor Portal instance could be successfully deployed into the COOL
- [x] The Assessor Portal web application could be successfully started and accessed locally from a browser on the Assessor Portal instance
- [x] Multiple Assessor Portal instances could be successfully deployed in the same COOL assessment environment
- [x] An operator on one Assessor Portal instance could NOT access a running Assessor Portal web application on a different Assessor Portal instance
- [x] The Assessor Portal web application (once started) could NOT be accessed from any other instances in the COOL assessment environment
- [x] The Assessor Portal web application (once started) could NOT be accessed externally via the Internet

## ✅ Checklist ##

* [x] This PR has an informative and human-readable title.
* [x] Changes are limited to a single goal - _eschew scope creep!_
* [x] All relevant type-of-change labels have been added.
* [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
* [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
* [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
* [x] All new and existing tests pass.

### CODE REVIEW

File: src/assessor_portal.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      ansible.builtin.lineinfile:
+        path: /lib/systemd/system/docker.service
+        backrefs: yes
+        # The exclamation mark forces us to quote the string because
+        # it is a special character in YAML.
+        #
+        # The (?!...) jazz is a negative look-ahead assertion.  See
+        # here, for example:
+        # https://docs.python.org/3/library/re.html
+        regexp: '^After=(.*)(?! multi\-user\.target cloud\-final\.service)$'
```
Comment (jsf9k) [inline src/assessor_portal.yml:39]: :muscle: regex-fu!

File: src/assessor_portal.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+#         when:
+#           - ansible_service_mgr == "systemd"
+#   - name: Create Assessor Portal desktop shortcut for VNC user
+#     block:
+#       - name: Create VNC user Desktop directory
+#         ansible.builtin.file:
+#           path: /home/{{ vnc_username }}/Desktop
+#           state: directory
+#           mode: 0755
+#           owner: "{{ vnc_username }}"
```
Comment (jsf9k) [inline src/assessor_portal.yml]: You should remove all this commented-out foo if you end up not using it.

File: src/packer.json [hunk 3]
```diff
@@ -29,13 +29,13 @@
...
         "filters": {
-          "name": "debian-10-amd64-*",
+          "name": "debian-11-amd64-*",
           "root-device-type": "ebs",
           "virtualization-type": "hvm"
         },
         "most_recent": true,
         "owners": [
-          "136693071363"
+          "903794441882"
```
Comment (jsf9k) [inline src/packer.json:38]: Eventually an official Debian 11 AMI will be released, at which point you will probably need to revert this owner account change.  I am now checking monthly that our base AMIs will still match the most recent (there is a recurring calendar item) so this should eventually get picked up by that.

**Review by jsf9k — COMMENTED** [review]
A few pre-review comments.  No changes requested.

File: src/assessor_portal.yml (hunk 2 above)
Comment (dav3r) [reply src/assessor_portal.yml]: I did end up removing the systemd bits as I don't think they are necessary for a minimum viable product here.

File: src/packer.json (hunk 3 above)
Comment (dav3r) [reply src/packer.json:38]: Good call - thanks for the reminder.  When I have a few minutes, I will create issues about that for this repo and for [cisagov/pca-gophish-composition-packer](https://github.com/cisagov/pca-gophish-composition-packer), which is in a similar situation.

**Review by jsf9k — APPROVED** [review]
Approved, but please create those issues that you mentioned in response to my earlier reviews.

File: src/packer.json (hunk 3 above)
Comment (dav3r) [reply src/packer.json:38]: Issues documented in https://github.com/cisagov/assessor-portal-packer/issues/2 and https://github.com/cisagov/pca-gophish-composition-packer/issues/56.

**Comment by dav3r** [comment]
> Approved, but please create those issues that you mentioned in response to my earlier reviews.

Issues documented in https://github.com/cisagov/assessor-portal-packer/issues/2 and https://github.com/cisagov/pca-gophish-composition-packer/issues/56.

File: README.md [hunk 4]
```diff
@@ -1,14 +1,9 @@
...
-new [cisagov](https://github.com/cisagov) GitHub
-[Packer](https://packer.io) project started. This skeleton project
-contains [licensing information](LICENSE), as well as
-[pre-commit hooks](https://pre-commit.com) and
-[GitHub Actions](https://github.com/features/actions) configurations
-appropriate for the major languages that we use.
+This repository is used to create images containing the Assessor Portal
+software.
 
 ## Pre-requisites ##
```
Comment (mcdonnnj) [inline README.md:9]: Please update the `Using Your Local Environment` section to include the `BUILD_BUCKET` environment variable as a requirement for the Packer template.

**Review by mcdonnnj — APPROVED** [review]
Strong work 💪 Please see my comment about a missing documentation update.

File: README.md (hunk 4 above)
Comment (dav3r) [reply README.md:9]: Good catch!  Rectified in https://github.com/cisagov/assessor-portal-packer/pull/1/commits/3fad09095574c0b01545002b0eeb24dd69640382.

**Pull request merged**